### PR TITLE
Various strictness improvements

### PR DESCRIPTION
**Pull request by wz1000**
Various strictness annotations that marginally improve heap usage.

### CODE REVIEW

File: ghcide/src/Development/IDE/Core/Compile.hs [hunk 1]
```diff
@@ -1455,18 +1455,6 @@ loadInterface session ms linkableNeeded RecompilationInfo{..} = do
 
     case (mb_checked_iface, recomp_iface_reqd) of
       (Just iface, UpToDate) -> do
-         -- If we have an old value, just return it
```
Comment (wz1000) [inline ghcide/src/Development/IDE/Core/Compile.hs:1458]: It turns out that reusing the same `ModDetails` as we had last time can keep the previous `HscEnv` and corresponding `HomePackageTable` data structures alive. Solution is to delete this code and always regenerate the `ModDetails` from the interface.

File: plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs [hunk 2]
```diff
@@ -45,7 +46,7 @@ data GetInstanceBindTypeSigs = GetInstanceBindTypeSigs
 
 data InstanceBindTypeSig = InstanceBindTypeSig
     { bindName     :: Name
-    , bindRendered :: T.Text
+    , bindRendered :: !T.Text
```
Comment (fendor) [inline plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs:49]: Any reason this needs to be strict while the other values arent?

Comment (wz1000) [reply plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs:49]: It showed up in a profile, the others are likely to already be evaluated I think.

Comment (fendor) [reply plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs:49]: Then a comment would be nice.
Also, would a guideline be wise that generally says everything in records needs a bang, and only in certain exceptional cases you don't need them?

**Review by fendor — APPROVED** [review]
LGTM

File: plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs (hunk 2 above)
Comment (michaelpj) [reply plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs:49]: Yeah, it seems clearer to bang everything in situations like this? Unsure.

File: ghcide/src/Development/IDE/Core/Compile.hs (hunk 1 above)
Comment (mpickering) [reply ghcide/src/Development/IDE/Core/Compile.hs:1457]: Yes!! Very important to do this (and cheap)

Comment (michaelpj) [reply ghcide/src/Development/IDE/Core/Compile.hs:1457]: Do we need a cautionary note? should we somehow get rid of `old_value` entirely? or is it okay to use it so long as we don't retain anything from it?

Comment (wz1000) [reply ghcide/src/Development/IDE/Core/Compile.hs:1457]: We need it for the old `ModIface` (which is safe to use).

**Pull request merged**